api/handlers/deployment.go

```diff
@@ -70,7 +70,7 @@ func (h *Deployment) create(r *http.Request) (*routing.Response, error) {
 
 	runnerInfo, err := h.runnerInfoRepo.GetRunnerInfo(r.Context(), authInfo, h.runnerName)
 	if err != nil {
-		return nil, apierrors.LogAndReturn(logger, err, "Error getting runner info in repository")
+		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Error getting runner info in repository")
```

Comment:
If I understand correctly, the user will receive a 404 back when they try to use rolling deploys and the runner doesn't support them. I'm not sure that is the correct error code, since it implies that the create endpoint doesn't exist. It would be better to return a 400 with a clear message explaining that rolling deploys aren't supported by this configuration. I'm open to other error codes, but I couldn't find a better fit.

Also, there don't appear to be any tests for this logic branch.

Reply:
I agree with Matt. I think `404` is better than a `500` (which is what this returned originally right?) and aligns with our other "not implemented" endpoints which just `404`, but the more I think about this it is implemented and in a mixed-runner case it could be supported for some apps but not others.

Reply:
I think we should definitely keep logging that this was because we couldn't fetch the `RunnerInfo` though.

Reply:
OK, we think we fixed it.